### PR TITLE
feat: language selector in navigation #P23-388

### PR DESCRIPTION
**Pull request by Alessiass**
## Jira 📖
https://tracker.intive.com/jira/browse/PATRO23-388

## What's new? ❓ 
User can now change default language during regflow. Code adjusted to utilize new atomic selector.

## Design 🎨 
Fully approved by our dear customer ❤️

### CODE REVIEW

File: apps/web/app/(regflow)/layout.styles.tsx [hunk 1]
```diff
@@ -119,13 +119,29 @@ export const ParagraphStyled = styled.p`
...
 `;
+
+export const LanguageSelectorWrapper = styled.div`
+  position: absolute;
+  top: 8px;
+  right: 8px;
+  z-index: 2;
+  border: 2px solid ${({ theme }) => theme.input.borderError};
+  border-radius: 8px;
+  padding: 8px;
```
Comment (wyola) [inline apps/web/app/(regflow)/layout.styles.tsx:141]: this can be removed when you apply variant styles in `LanguageSelector`

**Review by wyola — COMMENTED** [review]
As a creator of the atomic select I felt obligated to comment on this one 😄
Good addition! ⭐  Few styling things below

**Review by zuzannapaw — COMMENTED** [review]
I have noticed that when I opened branch for the first time, I didn't have any visible text on page. Only after choosing some language from select, my choice has been remembered and it worked properly (as well when I was closing and reopening app). It looked like there was no default value for language, like in our case I believe english is default. I think it is worth to look at it 👀 🤔 But anyway, great job with this task 💪

File: apps/web/app/(regflow)/layout.tsx [hunk 2]
```diff
@@ -69,7 +71,12 @@ export default function RegFlowLayout({ children }: LayoutProps) {
           </TypoStyled>
         </SectionStyled>
         <FormWrapperStyled>
-          <CustomCard>{children}</CustomCard>
+          <CustomCard>
+            <LanguageSelectorWrapper>
+              <LanguageSelector variant={"descriptive"} />
```
[the remark this answers is not in the document]
Comment (Alessiass) [reply apps/web/app/(regflow)/layout.tsx]: Fixed 🚀

File: apps/web/components/Navbar/LanguageSelector.tsx [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
-
 import { Flag, Select } from "ui";
 import { SelectLabelHiddenInTrigger } from "ui/Select/Select.styles";
+import styled from "styled-components";
+
+type LanguageSelectorProps = {
+  variant: "flag" | "descriptive";
+};
 
 const SelectStyled = styled(Select)`
```
[the remark this answers is not in the document]
Comment (Alessiass) [reply apps/web/components/Navbar/LanguageSelector.tsx]: Thanks :D Nice refactor 🚀

File: apps/web/app/(regflow)/layout.styles.tsx (hunk 1 above)
Comment (Alessiass) [reply apps/web/app/(regflow)/layout.styles.tsx]: Done 👍

File: apps/web/components/Navbar/LanguageSelector.tsx [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 import { useHasScrollBar } from "lib/hooks/useHasScrollBar";
 import { useTranslate } from "lib/hooks";
-import styled from "styled-components";
-
 import { Flag, Select } from "ui";
 import { SelectLabelHiddenInTrigger } from "ui/Select/Select.styles";
+import styled from "styled-components";
+
+type LanguageSelectorProps = {
+  variant: "flag" | "descriptive";
```
[the remark this answers is not in the document]
Comment (Alessiass) [reply apps/web/components/Navbar/LanguageSelector.tsx]: Fixed, I don't know why I always forget about it :D I promise, it's the last time 🙏

**Review by monika-klokosinska-intive — APPROVED** [review]

**Review by PawelPazderski — APPROVED** [review]
Nice and useful feature, looks good and works well 🎸 
One thing that caught my eye not related to the task itself - at 375px wide the card goes out of width 👇 in sign in and sign up

image

**Pull request merged**